src/sklearn_compat/_sklearn_compat.py

```diff
@@ -355,17 +356,167 @@ def _construct_instances(Estimator):
         yield _construct_instance(Estimator)
 
     # validation
-    def validate_data(_estimator, /, **kwargs):
-        if "ensure_all_finite" in kwargs:
-            force_all_finite = kwargs.pop("ensure_all_finite")
+    def validate_data(
+        _estimator,
+        /,
+        X="no_validation",
+        y="no_validation",
+        reset=True,
+        validate_separately=False,
+        skip_check_array=False,
+        **kwargs,
+    ):
+        """Validate input data and set or check feature names and counts of the input.
```

Comment:
I think it makes sense to keep this file short and refer to sklearn docs instead.